CMakeLists.txt

```diff
@@ -7,7 +7,7 @@ if (MSVC AND MSVC_VERSION VERSION_LESS "1900")
     return()
 endif()
 
-find_package(LLVM REQUIRED)
+find_package(LLVM REQUIRED CONFIG PATHS ${CMAKE_PREFIX_PATH} /opt/rocm )
```

Comment:
OS-specific code should go separately.

Reply:
@emankov : you mean /opt/rocm? Let me update

Reply:
@emankov : Updated new commit

Reply:
I meant `if (UNIX)`, actually

Reply:
@emankov, the current code : find_package(LLVM REQUIRED CONFIG PATHS ${CMAKE_PREFIX_PATH}) , is  OS agnostic now.
This shall run perfectly fine on any OS. Its is just trying to search as given by the CMAKE_PREFIX_PATH while compiling. It also shall search for the usual locations as well.
The last code was specific to rocm and was relying that the rocm is installed on the system (in linux /opt/rocm) hence it would get the path from there.
Request you to kindly review.

Reply:
Yes, but initially you wanted a default value for cases when `CMAKE_PREFIX_PATH` is not provided, right? Why did you decide to remove it?

Reply:
Yes, because you reviewed that code should be OS agnostic.  latest commit new code is adapting both now

Reply:
I see `if (NOT ...)` you like more. Well, do not mind, but `if (...)` is shorter, nonetheless.